src/dotnet/Chat.UI.Blazor/Components/PlaceSettings/PlaceSettingsEditTypeModalPage.razor

```diff
@@ -0,0 +1,224 @@
+                    </Content>
+                </TileItem>
+            </FormBlock>
+        }
+    } else if (m.Place.Rules.CanInvite()) {
+        <FormBlock Title="Manage links" Class="first last">
+            @if (m.Invites.Count > 0) {
+                <InviteList
+                    Invites="m.Invites"
+                    LinkFormat="@InviteLinkFormat.PrivateChat"
+                    NewInviteId="@_newInviteId"
+                    AllowRevoke="true"/>
+            }
+            <Button IsDisabled="@(m.Invites.Count >= MaxInvitesCount)" Click="OnNewInviteClick" Class="btn-modal h-10 btn-w-full">
+                <Icon><i class="icon-plus text-xl"></i></Icon>
+                <Title>New private join link</Title>
+            </Button>
+        </FormBlock>
+    }
+</Form>
+
+
+@code {
+    private const int MaxInvitesCount = 5;
+    private Form?_formRef;
+    private FormModel? _form;
+    private Symbol _newInviteId = Symbol.Empty;
+    private DialogButtonInfo _submitButtonInfo = null!;
+
+    [Inject] private ChatUIHub Hub { get; init; } = null!;
+    [Inject] private ComponentIdGenerator ComponentIdGenerator { get; init; } = null!;
+    [Inject] private DiffEngine DiffEngine { get; init; } = null!;
+
+    private Session Session => Hub.Session();
+    private IPlaces Places => Hub.Places;
+
+    [CascadingParameter] public DiveInModalPageContext Context { get; set; } = null!;
+
+    private PlaceId PlaceId { get; set; }
+
+    protected override void OnInitialized() {
+        PlaceId = Context.GetTypedModel<PlaceId>();
+        Context.SetTitle("Edit place settings");
+        _submitButtonInfo = DialogButtonInfo.CreateSubmitButton("Save", OnSubmit);
+        Context.RegisterButtons(DialogButtonInfo.CancelButton, _submitButtonInfo);
+        base.OnInitialized();
+    }
+
+    protected override async Task OnInitializedAsync() {
+        var place = await Places.Get(Session, PlaceId, default).Require();
+        _form = new FormModel(ComponentIdGenerator) {
+            IsPublic = place.IsPublic,
+        };
+        Context.SetClass("edit-chat-type");
+        await base.OnInitializedAsync();
+    }
+
+    protected override ComputedState<ComputedModel>.Options GetStateOptions()
+        => ComputedStateComponent.GetStateOptions(GetType(),
+            static t => new ComputedState<ComputedModel>.Options() {
+                InitialValue = ComputedModel.Loading,
+                Category = ComputedStateComponent.GetStateCategory(t),
+            });
+
+    protected override async Task<ComputedModel> ComputeState(CancellationToken cancellationToken) {
+        var placeId = PlaceId;
+
+        var place = await Places.Get(Session, placeId, default).Require();
+        if (!place.Rules.CanEditProperties())
+            return new() { Place = place };
+
+        var invites = await Hub.Invites.ListPlaceInvites(Session, placeId, cancellationToken);
```

Comment:
Add shorcut properties for hub properties: Hub.Invites, Hub.Clocks, Hub.UICommander()

Reply:
Done